runtime/stdlib/account.go

```diff
@@ -791,6 +800,245 @@ func newPublicAccountContractsValue(
+				})
+			}
+
+			handler.EmitEvent(
+				inter,
+				AccountInboxRemoveEventType,
+				[]interpreter.Value{
+					providerValue,
+					providerValue,
+					nameValue,
+				},
+				getLocationRange,
+			)
+
+			inter.WriteStored(address, inboxStorageDomain, valuePath(nameValue.Str), nil)
+			inter.WriteStored(address, inboxStorageDomain, recipientPath(nameValue.Str), nil)
+
+			publishedValue = publishedValue.Transfer(
+				inter,
+				getLocationRange,
+				atree.Address(address),
+				true,
+				nil,
+			)
+
+			return interpreter.NewSomeValueNonCopying(inter, publishedValue)
+		},
+		sema.AuthAccountTypeInboxPublishFunctionType,
+	)
+}
+
+func accountInboxClaimFunction(
+	gauge common.MemoryGauge,
+	handler EventEmitter,
+	address common.Address,
+	recipientValue interpreter.AddressValue,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewHostFunctionValue(
+		gauge,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			nameValue, ok := invocation.Arguments[0].(*interpreter.StringValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			providerValue, ok := invocation.Arguments[1].(interpreter.AddressValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			inter := invocation.Interpreter
+			getLocationRange := invocation.GetLocationRange
+
+			providerAddress := providerValue.ToAddress()
+
+			publishedValue := inter.ReadStored(providerAddress, inboxStorageDomain, valuePath(nameValue.Str))
+
+			if publishedValue == nil {
+				return interpreter.NewNilValue(gauge)
+			}
+
+			// compare the intended recipient with the caller
+			intendedRecipient, ok := inter.ReadStored(providerAddress, inboxStorageDomain, recipientPath(nameValue.Str)).(interpreter.AddressValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+			if intendedRecipient.ToAddress() != recipientValue.ToAddress() {
+				return interpreter.NewNilValue(gauge)
+			}
+
+			typeParameterPair := invocation.TypeParameterTypes.Oldest()
+			if typeParameterPair == nil {
+				panic(errors.NewUnreachableError())
+			}
+
+			ty := sema.NewCapabilityType(gauge, typeParameterPair.Value)
+			publishedType := publishedValue.StaticType(invocation.Interpreter)
+			if !inter.IsSubTypeOfSemaType(publishedType, ty) {
+				panic(interpreter.ForceCastTypeMismatchError{
+					ExpectedType:  ty,
+					ActualType:    inter.MustConvertStaticToSemaType(publishedType),
+					LocationRange: getLocationRange(),
+				})
+			}
+
+			inter.WriteStored(providerAddress, inboxStorageDomain, valuePath(nameValue.Str), nil)
+			inter.WriteStored(providerAddress, inboxStorageDomain, recipientPath(nameValue.Str), nil)
+
+			handler.EmitEvent(
+				inter,
+				AccountInboxRemoveEventType,
+				[]interpreter.Value{
+					providerValue,
+					recipientValue,
+					nameValue,
+				},
+				getLocationRange,
+			)
+
+			publishedValue = publishedValue.Transfer(
+				inter,
+				getLocationRange,
+				atree.Address(address),
+				true,
+				nil,
+			)
```

Comment:
Same as above: first transfer from account to to stack (empty) address, then write nil t storage.